Display the rerrors of multiple analysis requests distinctly

`doAnalysis` clobbers together the results of all the analysis requests, and errors can not be displayed with their associated analysis requests.

Print errors and results distinctly, with information about the analysis request next to the error or result itself.

@nbanmp  thanks  for raising this issue.  If you think this is much convenient way to show results to user then I'd be glad to help. But  there are few issues you might run into.

Becuase one solidity file might contain multiple contracts truffle outputs compile results into separate json files. Then we use details of contract to MythX API. So what happens is that you have multiple responses pointing to same solidity file. truffle-security output will look like this:
```
/home/daniyar/tmp/truffle-security-test/contracts/03-auction.sol
   1:0   warning  A floating pragma is set                       SWC-103
   5:2   warning  State variable shadows another state variable  SWC-119
   5:10  warning  The state variable visibility is not set       SWC-108
   6:2   warning  State variable shadows another state variable  SWC-119
   6:7   warning  The state variable visibility is not set       SWC-108
   9:2   warning  The function visibility is not set             SWC-100
  19:4   error    persistent state read after call               SWC-107
  20:4   error    persistent state write after call              SWC-107
  27:10  warning  The state variable visibility is not set       SWC-108
  28:10  warning  The state variable visibility is not set       SWC-108
  30:27  warning  The state variable visibility is not set       SWC-108
  51:4   warning  Unchecked return value from low level call     SWC-104

/home/daniyar/tmp/truffle-security-test/contracts/03-auction.sol
   1:0   warning  A floating pragma is set                           SWC-103
   5:2   warning  State variable shadows another state variable      SWC-119
   5:10  warning  The state variable visibility is not set           SWC-108
   6:2   warning  State variable shadows another state variable      SWC-119
   6:7   warning  The state variable visibility is not set           SWC-108
   9:2   warning  The function visibility is not set                 SWC-100
  27:10  warning  The state variable visibility is not set           SWC-108
  28:10  warning  The state variable visibility is not set           SWC-108
  30:27  warning  The state variable visibility is not set           SWC-108
  51:4   warning  Unchecked return value from low level call         SWC-104
  51:4   warning  The return value of a message call is not checked  SWC-104

✖ 23 problems (2 errors, 21 warnings)
```
Because we have multiple analysis results pointing to same file and  duplicate SWC issues we group analysis results and remove duplicate entries.
I think it would be great to work on output format as well. Even come up with custom formatter style. By default eslint's stylish formatter is used.
What do you think?

--- a/helpers.js
+++ b/helpers.js
@@ -297,7 +297,7 @@ const doAnalysis = async (client, config, jsonFiles, contractNames = null, limit
                 });
                 bar.terminate();    // terminate since bar.complete is false at this time
             }
-            return [err, null];
+            return [(buildObj.contractName + ": ").yellow + err, null];
         }
     });
 
@@ -331,7 +331,7 @@ function doReport(config, objects, errors, notFoundContracts) {
     }
 
     if (errors.length > 0) {
-        config.logger.error('Internal MythX errors encountered:');
+        config.logger.error('Internal MythX errors encountered:'.red);
         errors.forEach(err => {
             config.logger.error(err.error || err);
             if (config.debug > 1 && err.stack) {

